shared/teams/main/index.js

```diff
@@ -5,62 +5,150 @@ import * as Kb from '../../common-adapters'
+  onJoinTeam: () => void,
+  onManageChat: string => void,
+  onOpenFolder: string => void,
+  onReadMore: () => void,
+  onViewTeam: string => void,
+  sawChatBanner: boolean,
+  teamNameToIsOpen: {[key: string]: boolean},
+  teammembercounts: {[key: string]: number},
+  teamnames: $ReadOnlyArray<string>,
+  teamresetusers: {[key: string]: Array<string>},
+  teamToRequest: {[key: string]: number},
+  title?: string,
 |}
 
-const Teams = (props: Props) => (
-  <Kb.Box style={{...Styles.globalStyles.flexBoxColumn, alignItems: 'center', height: '100%'}}>
-    <Header loaded={props.loaded} onCreateTeam={props.onCreateTeam} onJoinTeam={props.onJoinTeam} />
-    <Kb.Box style={{flex: 1, position: 'relative', width: '100%'}}>
-      <Kb.ScrollView
-        style={{...Styles.globalStyles.fillAbsolute}}
-        contentContainerStyle={{
-          ...Styles.globalStyles.flexBoxColumn,
-          alignItems: 'center',
-        }}
-      >
-        {!props.sawChatBanner && (
-          <Banner onReadMore={props.onReadMore} onHideChatBanner={props.onHideChatBanner} />
-        )}
-        <TeamList
-          teamnames={props.teamnames}
-          teammembercounts={props.teammembercounts}
-          teamresetusers={props.teamresetusers}
-          teamNameToIsOpen={props.teamNameToIsOpen}
-          newTeams={props.newTeams}
-          newTeamRequests={props.newTeamRequests}
-          onOpenFolder={props.onOpenFolder}
-          onManageChat={props.onManageChat}
-          onViewTeam={props.onViewTeam}
-        />
-        <BetaNote onReadMore={props.onReadMore} />
-        {/* Put progress indicator in the footer on mobile because it won't fit in the header on small screens */}
-        {Styles.isMobile && (
-          <Kb.ProgressIndicator
-            style={{
-              alignSelf: 'center',
-              marginBottom: Styles.globalMargins.small,
-              opacity: props.loaded ? 0 : 1,
-              width: 20,
-            }}
+type RowProps = {
+  firstItem: boolean,
+  name: string,
+  membercount: number,
+  isNew: boolean,
+  isOpen: boolean,
+  newRequests: number,
+  onOpenFolder: () => void,
+  onManageChat: () => void,
+  resetUserCount: number,
+  onViewTeam: () => void,
+}
+
+export const TeamRow = React.memo<RowProps>((props: RowProps) => {
+  const badgeCount = props.newRequests + props.resetUserCount
+
+  return (
+    <Kb.ListItem2
+      type="Large"
+      firstItem={props.firstItem}
+      onClick={props.onViewTeam}
+      icon={
+        <Kb.Box2 direction="vertical" style={styles.avatarContainer}>
+          <Kb.Avatar size={Styles.isMobile ? 48 : 32} teamname={props.name} isTeam={true} />
+          {!!badgeCount && <Kb.Badge badgeNumber={badgeCount} badgeStyle={styles.badge} />}
+        </Kb.Box2>
+      }
+      body={
+        <Kb.Box2 direction="vertical" fullWidth={true}>
+          <Kb.Box2 direction="horizontal" gap="tiny" alignSelf="flex-start">
+            <Kb.Text type="BodySemibold">{props.name}</Kb.Text>
```

Comment:
`lineClamp` / more right padding here for long teamnames? 
<img width="409" alt="image" src="https://user-images.githubusercontent.com/11968340/53043402-64885980-3456-11e9-9753-1b8b4cb1a00b.png">